src/components/scenes/WcConnectScene.js

```diff
@@ -126,7 +125,7 @@ export const WcConnectScene = (props: Props) => {
       return <SelectableRow onPress={showWalletListModal} title={s.strings.wc_confirm_select_wallet} arrowTappable />
     } else {
       const walletNameStr = truncateString(walletName || '', MAX_ADDRESS_CHARACTERS)
-      const walletImage = <FastImage style={styles.currencyLogo} source={{ uri: walletImageUri }} />
+      const walletImage = <CurrencyIcon tokenId={contractAddress} walletId={selectedWallet.walletId} resizeMode="contain" />
```

Comment:
Please do not pass the contract address as the tokenId. I don't want landmines scattered throughout the program. Only the currency plugin has the ability to pick a tokenId. Everyone else has to look it up from the core.

Reply:
It's temporary. 
We need to update everywhere to use pluginId and tokenId instead of currencyCode and contractAddress.
So for now it works but we would change it very soon.